File: pocs/android_only_m2_j17jars/app/src/main/java/bisq/android/main/user_profile/UserProfileModel.java
package bisq.android.main.user_profile;

import java.security.KeyPair;

import bisq.common.data.ByteArray;
import bisq.common.observable.Observable;
import bisq.security.pow.ProofOfWork;
import bisq.user.identity.UserIdentity;
import lombok.Getter;
import lombok.Setter;

@Getter
public class UserProfileModel {
    @Setter
    private KeyPair keyPair;
    @Setter
    private byte[] pubKeyHash;
    @Setter
    private ProofOfWork proofOfWork;

    private final Observable<String> userName = new Observable<>();
    private final Observable<String> terms = new Observable<>("");
    private final Observable<String> statement = new Observable<>("");
    private final Observable<String> nym = new Observable<>();
    private final Observable<String> nickName = new Observable<>();
    private final Observable<String> profileId = new Observable<>();

    private final Observable<Boolean> isBusy = new Observable<>();
}

File: pocs/android_only_m2_j17jars/app/src/main/java/bisq/android/main/user_profile/UserProfileModel.kt
package bisq.android.main.user_profile

import bisq.common.observable.Observable
import bisq.security.pow.ProofOfWork
import lombok.Getter
import lombok.Setter
import java.security.KeyPair

@Getter
class UserProfileModel {
    @Setter
    var keyPair: KeyPair? = null

    @Setter
    lateinit var pubKeyHash: ByteArray

    @Setter
    var proofOfWork: ProofOfWork? = null

    private val userName = Observable<String>()
    val terms = Observable("")
    val statement = Observable("")
    val nym = Observable<String>()
    private val nickName = Observable<String>()
    private val profileId = Observable<String>()

    val isBusy = Observable<Boolean>()
}
